src/js/shepherd.js

```diff
@@ -531,8 +531,12 @@ class Step extends Evented {
       content.appendChild(footer);
     }
 
-    document.body.appendChild(this.el);
-
+    // Handling open dialog elements.
+    if (document.querySelector('dialog[open]')) {
```

Comment:
I'm not super familiar with dialog elements, but wouldn't it be possible someone could have one open, but still want to append to the body instead?

Reply:
Normally open dialog elements having the main focus and getting closed after an specific action is performed like window.confirm() or window.promt().
Example: https://material-components.github.io/material-components-web-catalog/#/component/dialog

Reply:
Right, but it is possible you would have an open dialog and still want to not attach your steps to it. I think a better solution to this problem might be to allow passing a context and default to `document.body` when one is not passed. What do you think?

Reply:
I see the general scenarios:
1. Main window has steps, Dialog has steps.
2. Main window has no steps, Dialog has steps.
3. Main window has steps, Dialog has no steps.

All three scenarios are covered in this case:
1. Steps are attached to body. Dialog gets open. Dialog steps are attached to the Dialog.
2. Dialog gets open. Dialog steps are attached to the Dialog.
3. Steps are attached to body.

Edge Case:
The dialog gets open before the main window adding the steps.
This should theoretical never happen, because such dialog normally only shown on user interaction or after the main content is rendered.

Did I miss anything?